godot_project/editor/rendering/atomic_structure_renderer/representation/springs_representation/springs_representation.gd

```diff
@@ -274,10 +281,12 @@ func handle_hover_structure_changed(_in_toplevel_hovered_structure_context: Stru
 		return
 	
 	if _hovered_spring != AtomicStructure.INVALID_SPRING_ID:
-		var color := COLOR_HIGHLIGHT if structure_context.is_spring_selected(_hovered_spring) else COLOR_LOWLIGHT
-		_spring_renderer.change_spring_color(_hovered_spring, color)
+		var is_hovered_selected: bool = structure_context.is_spring_selected(_hovered_spring)
```

Comment:
nit: this spring is no longer hovered, so the variable name `is_hovered_selected` is missleading

Reply:
hmm I don't see anything that would suggest to me it's no longer hovered, so it seems I fell into that trap. 
I think if it would not be hovered the id stored inside `_hovered_spring` would be equal to `INVALID_SPRING_ID` which clearly is not the case here

Reply:
what the code is doing is "unhovering" the previously hovered spring before hovering the new one, if the spring remains the same, it makes changes for no good reason, gut the point is that _hovered_spring is about to change at the end of the callback so what is stored in _hovered_spring is no longer the _hovered_spring (is also the case when unhovering the spring and now no spring is hovered)